pkl-core/src/main/java/org/pkl/core/messaging/Message.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.messaging;
+
+public interface Message {
+
+  Type getType();
+
+  enum Type {
+    CREATE_EVALUATOR_REQUEST(0x20),
+    CREATE_EVALUATOR_RESPONSE(0x21),
+    CLOSE_EVALUATOR(0x22),
+    EVALUATE_REQUEST(0x23),
+    EVALUATE_RESPONSE(0x24),
+    LOG_MESSAGE(0x25),
+    READ_RESOURCE_REQUEST(0x26),
+    READ_RESOURCE_RESPONSE(0x27),
+    READ_MODULE_REQUEST(0x28),
+    READ_MODULE_RESPONSE(0x29),
+    LIST_RESOURCES_REQUEST(0x2a),
+    LIST_RESOURCES_RESPONSE(0x2b),
+    LIST_MODULES_REQUEST(0x2c),
+    LIST_MODULES_RESPONSE(0x2d),
+    INITIALIZE_MODULE_READER_REQUEST(0x100),
```

Comment:
I think this was to hard-differentiate between server messages and external reader messages, based on one of our discussions quite a while back. Definitely open to revising this.

Reply:
Yeah, since many of these messages are shared between the two different flows, I'm not sure if it matters that much.

Maybe better to just keep this incremental.

Reply:
Updated!